lib/IRGen/GenKeyPath.cpp

```diff
@@ -1105,6 +1106,39 @@ emitKeyPathComponent(IRGenModule &IGM,
   case KeyPathPatternComponent::Kind::OptionalWrap:
     fields.addInt32(KeyPathComponentHeader::forOptionalWrap().getData());
     break;
+  case KeyPathPatternComponent::Kind::TupleElement:
+    assert(baseTy->is<TupleType>() && "not a tuple");
+
+    SILType loweredTy = IGM.getSILTypes().getLoweredType(baseTy);
+
+    // Tuple with fixed layout
+    //
+    // This code is ALSO executed in the case of a tuple with dynamic layout,
+    // (see below) but only if `component.getTupleIndex()` is 0 - in that case
+    // the compiler knows that the tuple element is always at offset 0.
+    // TODO: If this is behavior is not desired we should find a way to skip to
+    // the next section of code e.g. check if baseTy has archetypes?
```

Comment:
@jckarter thoughts about this comment?

Reply:
This sounds reasonable. I don't think we'd ever put the first tuple element at a different offset.